### PR TITLE
Clear received header values on redirect

### PR DESCRIPTION
**Pull request by dennisschagt**
Fixes https://github.com/newsboat/newsboat/issues/2230 (at least part of it, I can't verify the roscosmos issues myself).

### CODE REVIEW

File: rss/parser.cpp [hunk 1]
```diff
@@ -75,7 +81,10 @@ static size_t handle_headers(void* ptr, size_t size, size_t nmemb, void* data)
 	const auto header = std::string(reinterpret_cast<const char*>(ptr), size * nmemb);
 	HeaderValues* values = static_cast<HeaderValues*>(data);
 
-	if (header.find("Last-Modified:") == 0) {
+	if (header.find("HTTP") == 0) {
```
Comment (der-lyse) [inline rss/parser.cpp:84]: Just an idea, but not sure if it were any better to check for `HTTP/` with a slash.

Comment (dennisschagt) [reply rss/parser.cpp]: I guess that might make it a lot less likely to incorrectly match on regular headers, so a good idea.
Thanks, done

File: rss/parser.cpp [hunk 2]
```diff
@@ -62,51 +62,57 @@ struct HeaderValues {
...
 				"handle_headers: last-modified %s "
 				"(curl_getdate "
 				"FAILED)",
-				header + 14);
+				header_value.c_str());
 		} else {
-			values->lastmodified =
-				curl_getdate(header + 14, nullptr);
+			values->lastmodified = r;
+			curl_getdate(header_value.c_str(), nullptr);
```
Comment (Minoru) [inline rss/parser.cpp:98]: This line is a leftover from editing and can probably be removed. I'll do it via GitHub suggestions;

```suggestion
```

File: rss/parser.cpp [hunk 3]
```diff
@@ -62,51 +62,57 @@ struct HeaderValues {
...
 			LOG(Level::DEBUG,
 				"handle_headers: got last-modified %s (%" PRId64 ")",
-				header + 14,
+				header_value.c_str(),
```
Comment (Minoru) [inline rss/parser.cpp:101]: `LOG` supports `std::string`, so no need to call `c_str()` here.
```suggestion
				header_value,
```

**Review by Minoru — APPROVED** [review]
Thanks! This indeed fixes the practicaltypography.com feed, but roscosmos still doesn't work. I'll re-open the relevant issue and work on it later.

**Pull request merged**